src/components/AvatarCropModal/AvatarCropModal.js

```diff
@@ -0,0 +1,312 @@
+
+    // A memoized by reanimated style, which updates when image size or scale changes
+    const imageStyle = useAnimatedStyle(() => {
+        const height = originalImageHeight.value;
+        const width = originalImageWidth.value;
+        const aspectRatio = height > width ? height / width : width / height;
+        const rotate = interpolate(rotation.value, [0, 360], [0, 360]);
+        return {
+            transform: [
+                {translateX: translateX.value},
+                {translateY: translateY.value},
+                {scale: scale.value * aspectRatio},
+                {rotate: `${rotate}deg`},
+            ],
+        };
+    }, [originalImageHeight.value, originalImageWidth.value, clamp]);
+
+    // Rotates the image by changing the rotation value by 90 degrees
+    // and updating the position so the image remains in the same place after rotation
+    const rotateImage = useCallback(() => {
+        rotation.value -= 90;
+
+        // Rotating 2d coordinates by applying the formula (x, y) → (-y, x).
+        [translateX.value, translateY.value] = [translateY.value, translateX.value * -1];
+
+        // Since we rotated the image by 90 degrees, now width becomes height and vice versa.
+        [originalImageHeight.value, originalImageWidth.value] = [
+            originalImageWidth.value,
+            originalImageHeight.value,
+        ];
+    }, []);
+
+    // Crops an image that was provided in the imageUri prop, using the current position/scale
+    // then calls onSave and onClose callbacks
+    const cropAndSaveImage = useCallback(() => {
+        const smallerSize = Math.min(originalImageHeight.value, originalImageWidth.value);
+        const size = smallerSize / scale.value;
+        const imageCenterX = originalImageWidth.value / 2;
+        const imageCenterY = originalImageHeight.value / 2;
+        const apothem = size / 2; // apothem for squares is equals to half of it size
+
+        // Since the translate value is only a distance from the image center, we are able to calculate
+        // the originX and the originY - start coordinates of cropping view.
+        const originX = imageCenterX - apothem - ((translateX.value / imageContainerSize / scale.value) * smallerSize);
+        const originY = imageCenterY - apothem - ((translateY.value / imageContainerSize / scale.value) * smallerSize);
+
+        const crop = {
+            height: size, width: size, originX, originY,
+        };
+
+        imageManipulator(props.imageUri, [{rotate: rotation.value % 360}, {crop}], {compress: 1})
+            .then((newImage) => {
+                props.onClose();
+                props.onSave(newImage);
+            });
+    }, [props.imageUri, imageContainerSize]);
+
+    return (
+        <Modal
+            onClose={props.onClose}
+            isVisible={props.isVisible}
+            shouldSetModalVisibility={props.shouldSetModalVisibility}
```

Comment:
Interesting, I don't see the prop definition for this. Also do we need this?

Reply:
Deleted